### ISSUE TITLE
BUG: Windows testing failure

### ISSUE THREAD

**Issue by rstoneback** [pysat/pysat#824]
**Describe the bug**
Unit tests on windows suddenly started failing.

**To Reproduce**
1. Check results on GitHub Actions

```
FAILED pysat/tests/test_utils.py::TestBasicNetCDF4::test_write_and_read_netcdf4_default_format_w_compression
213
FAILED pysat/tests/test_utils.py::TestBasicNetCDF4::test_write_and_read_netcdf4_default_format_higher_order_w_zlib
214
```

**Expected behavior**
These tests used to pass. My guess is something with zlib?

**Screenshots**

**Desktop (please complete the following information):**
 - OS: GitHub Actions computer
 
**Additional context**

**Comment by jklenzing** [comment]
netCDF4 1.5.7 was released a few days ago.  Looking through the logs, this seems to correspond to when the tests stopped working.  The changelog there includes an update to how netcdf4 runs on windows:  https://github.com/Unidata/netcdf4-python/blob/v1.5.7rel/Changelog

**Comment by jklenzing** [comment]
To troubleshoot, we could try a branch with a version cap to see if the version update is the cause.

**Comment by rstoneback** [comment]
Thanks @jklenzing. I'll work on this and the penumbra packages for the RC tomorrow.

**Comment by rstoneback** [comment]
#826 passes now but I think the netCDF folks may have updated things on their end before I got there. This is based on running pysat v3.0.1 integration tests on pysat penumbra packages.

**Comment by rstoneback** [comment]
penumbra packages have started failing on windows with netCDF. Looking like the version is the issue.

**Comment by jklenzing** [comment]
Depending on whether the penumbra packages use this aspect of netcdf4, it may or may not fail.  pysatNASA seems to be doing fine without a version limit.

**Comment by rstoneback** [comment]
Here is what I'm getting with pysatModels,
```
>   ???
264
E   OSError: [Errno -101] NetCDF: HDF error: b'/tmp/tmp0ezp5rmi/gnss/tec/vtec/gps171119g.002.netCDF4'
265
```

**Comment by rstoneback** [comment]
A version limit was added to current pysat to shield users from the issue. Not a solution, but an effective temporary patch. #826

**Comment by jklenzing** [comment]
netCDF4 1.5.8 fixes the windows issues in CI testing.  It was released Oct 31, 2021.  I've got it and other changes loaded into #967.  Not sure if we consider this issue fixed in addition to #828 with that pull.

**Comment by jklenzing** [comment]
Calling this fixed with #967 and release of netCDF4 1.5.8.
